tests/devapps/Intune-xamarin-ios/MainViewController.cs

```diff
@@ -0,0 +1,252 @@
+            string[] Scopes = { "https://xamarintruemamenterpriseapp-msidlab20.msappproxy.net//Hello.World" };
+            string[] clientCapabilities = { "ProtApp" };
+
+            if (useLab4)
+            {
+                // for IDLABTRUEMAMCA@msidlab4.onmicrosoft.com
+                // com.xamarin.microsoftintunemamsample
+                // Redirect URI = msauth.com.xamarin.microsoftintunemamsample://auth
+                // MSAL config
+                // let kClientID = "cc2ef30a-30ff-404e-b0c8-8022dc941b51"
+                // let kRedirectUri = "msauth.com.xamarin.microsoftintunemamsample://auth"
+                // let kAuthority = "https://login.microsoftonline.com/common"
+                // let kGraphEndpoint = "https://graph.microsoft.com/"
+
+                // string clientId = "cc2ef30a-30ff-404e-b0c8-8022dc941b51"; // my app in the lab
+                // string[] Scopes = { "User.Read" };
+            }
+
+            // IDLAB20TrueMAMCA@msidlab20.onmicrosoft.com
+            if (useLab20)
+            {
+                clientId = "6d50af5d-2529-4ff4-912f-c1d6ad06953e"; // my app in the lab
+                redirectURI = $"msauth.com.xamarin.microsoftintunemamsample://auth";
+                tenantID = "7257a09f-53cc-4a91-aca8-0cb6713642a5";
+                Scopes[0] = "api://09aec9b9-0b0f-488a-81d6-72fd13a3a1c1/Hello.World"; // needs admin consent
+            }
+
+            try
+            {
+                if (PCA == null)
+                {
+                    var pcaBuilder = PublicClientApplicationBuilder.Create(clientId)
+                                                                        .WithRedirectUri(redirectURI)
+                                                                        .WithIosKeychainSecurityGroup("com.microsoft.adalcache")
+                                                                        .WithLogging(MSALLogCallback, LogLevel.Verbose)
+                                                                        .WithTenantId(tenantID)
+                                                                        .WithClientCapabilities(clientCapabilities)
+                                                                        .WithHttpClientFactory(new HttpSnifferClientFactory())
+                                                                        .WithBroker(true);
+                    PCA = pcaBuilder.Build();
+                    var interParamBuilder = PCA.AcquireTokenInteractive(Scopes)
+                                                .WithParentActivityOrWindow(this)
+                                                .WithUseEmbeddedWebView(true);
+
+                    var authResult = await interParamBuilder.ExecuteAsync().ConfigureAwait(false);
+                    ShowAlert("Success Interactive 1", authResult.AccessToken);
+                }
+            }
+            catch (IntuneAppProtectionPolicyRequiredException ex)
```

Comment:
if ATS is done before ATI and the device is already enrolled, it will not call Intune to demo the desired functionality. So, the user will see that only for the first time and wonder why it does not work. May be there should be a comment indicating it.